src/main/java/me/desht/pneumaticcraft/common/thirdparty/PneumaticcraftIMC.java

```diff
@@ -0,0 +1,40 @@
+package me.desht.pneumaticcraft.common.thirdparty;
+
+import me.desht.pneumaticcraft.api.lib.Names;
+import me.desht.pneumaticcraft.lib.Log;
+import me.desht.pneumaticcraft.lib.ModIds;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.InterModComms;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.lifecycle.InterModEnqueueEvent;
+
+import java.util.ArrayList;
+
+/**
+ * Class containing manager for sending all IMC messages to other mods
+ */
+@Mod.EventBusSubscriber(modid = Names.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
+public class PneumaticcraftIMC {
```

Comment:
I presume this a message queue to collect all the messages we need to send at the right time?  If so, I'd suggest an `ArrayDeque` rather than `ArrayList`, just for tidiness (no need to keep queued messages around after they've been sent)

Reply:
I changed it, but out of curiosity, why is an `ArrayDeque` better than an `ArrayList` here?

Reply:
I mean, an `ArrayList` is OK, but I like to fit the data structure to the task.  And the task here is a queue of message, so `ArrayDeque` is just a more natural fit.

Reply:
Also means code like this can be used:
```java
while (!iMCMessageQueue.isEmpty()) {
    InterModComms.IMCMessage message = iMCMessageQueue.pop();
    InterModComms.sendTo(Names.MOD_ID, message.modId(), message.method(), message.messageSupplier());
}
```

And it occurred to me after that this can be called off-thread (from `FMLCommonSetupEvent`) so I made it a `ConcurrentLinkedQueue`.  A bit paranoid, but never a bad idea to be over-careful when it comes to threading.